plugins/woocommerce/tests/php/includes/wc-core-functions-test.php

```diff
@@ -224,4 +224,53 @@ function () {
 		$this->assertEquals( '', $result['state'] );
 		remove_all_filters( 'woocommerce_geolocate_ip' );
 	}
+
+	/**
+	 * Test wc_delete_transients() function.
+	 */
+	public function test_wc_delete_transients() {
+		// Set up test transients.
+		$transient_name1 = 'wc_test_transient_1';
+		$transient_name2 = 'wc_test_transient_2';
+
+		set_transient( $transient_name1, 'test_value_1', HOUR_IN_SECONDS );
+		set_transient( $transient_name2, 'test_value_2', HOUR_IN_SECONDS );
+
+		// Verify transients exist before deletion.
+		$this->assertEquals( 'test_value_1', get_transient( $transient_name1 ) );
+		$this->assertEquals( 'test_value_2', get_transient( $transient_name2 ) );
+
+		// Delete the transients.
+		wc_delete_transients( array( $transient_name1, $transient_name2 ) );
+
+		// Force WordPress to refresh its memory cache.
+		wp_cache_flush();
+
+		// Verify transients are deleted.
+		$this->assertFalse( get_transient( $transient_name1 ) );
+		$this->assertFalse( get_transient( $transient_name2 ) );
+
+		// Test with a single transient name (string instead of array).
+		$transient_name3 = 'wc_test_transient_3';
+		set_transient( $transient_name3, 'test_value_3', HOUR_IN_SECONDS );
+		$this->assertEquals( 'test_value_3', get_transient( $transient_name3 ) );
+
```

Comment:
Maybe also testing passing a single transient as string too before mapping it to an array?
```suggestion
$this->assertFalse( wc_delete_transients( $transient_name3 ) );
```

Reply:
Good idea! Added in https://github.com/woocommerce/woocommerce/pull/55931/commits/1eb53749c418a925c9ef1ef0355f5a4b7eb3fd17